### PR TITLE
Added error when dispute stake is over MAX

### PR DESCRIPTION
**Pull request by bthaile**
[Clubhouse Story](https://app.clubhouse.io/augur/story/8843)

## Submitter checklist
- [x] Test covering functionality added/fixed in
- Check the linked story includes the following:
  - [x] Steps to reproduce
  - [ ] Screenshot (If applicable)

## Reviewer Checklist
- [ ] Test/lint pass

### CODE REVIEW

File: test/reporting/actions/submit-market-contribute-test.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ describe(`modules/reporting/actions/submit-market-contribute.js`, () => {
   }
 
   const getPayoutNumerators = sinon.stub().returns([10000, 0])
-  const removeAccountDispute = sinon.spy()
+  const removeAccountDispute = sinon.stub().returns({ type: 'REMOVE_ACCOUNT_DISPUTE' })
```
Comment (justinbarry) [inline test/reporting/actions/submit-market-contribute-test.js]: Why was this needed? I'm not seeing anything asserted against it.

File: src/modules/reporting/components/reporting-dispute-form/reporting-dispute-form.jsx [hunk 2]
```diff
@@ -26,9 +26,15 @@ export default class ReportingDisputeForm extends Component {
     accountDisputeData: PropTypes.object,
   }
 
-  static checkStake(stake, updatedValidations) {
-    if (stake === '' || stake == null || stake <= 0) {
+  static checkStake(wholeREPstake, updatedValidations, maxRep) {
+    const bnMaxRep = createBigNumber(formatNumber(createBigNumber(maxRep).toNumber(), { decimals: 4, roundUp: true }).formattedValue)
+    const bnStake = createBigNumber(wholeREPstake || 0)
+    console.log('max rep', bnMaxRep.toNumber())
```
Comment (justinbarry) [inline src/modules/reporting/components/reporting-dispute-form/reporting-dispute-form.jsx]: Where are re: leaving console statements in the code? Do we have a policy?

File: src/modules/reporting/components/reporting-dispute-form/reporting-dispute-form.jsx [hunk 3]
```diff
@@ -26,9 +26,15 @@ export default class ReportingDisputeForm extends Component {
     accountDisputeData: PropTypes.object,
   }
 
-  static checkStake(stake, updatedValidations) {
-    if (stake === '' || stake == null || stake <= 0) {
+  static checkStake(wholeREPstake, updatedValidations, maxRep) {
```
Comment (justinbarry) [inline src/modules/reporting/components/reporting-dispute-form/reporting-dispute-form.jsx]: Given this is static could we move is to another file and add tests?

File: src/modules/reporting/components/reporting-dispute-form/reporting-dispute-form.jsx [hunk 4]
```diff
@@ -258,7 +265,7 @@ export default class ReportingDisputeForm extends Component {
       }
     }
 
-    ReportingDisputeForm.checkStake(stake, updatedValidations)
+    ReportingDisputeForm.checkStake(stake, updatedValidations, this.calculateMaxRep())
```
Comment (justinbarry) [inline src/modules/reporting/components/reporting-dispute-form/reporting-dispute-form.jsx]: Is it possible for this.state.selectedOutcome to exist here? If so, should it be passed in?

File: src/modules/reporting/components/reporting-dispute-form/reporting-dispute-form.jsx (hunk 3 above)
Comment (bthaile) [reply src/modules/reporting/components/reporting-dispute-form/reporting-dispute-form.jsx]: could, but it's not used by anything else. It has errors that are specifically for this form. I want to keep it with the form.

**Comment by bthaile** [comment]
@justinbarry Changes made, should be gtg.
